osu.Game/Overlays/Profile/Sections/Kudosu/DrawableKudosuHistoryItem.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+
+namespace osu.Game.Overlays.Profile.Sections.Kudosu
+{
+    public class DrawableKudosuHistoryItem : CompositeDrawable
+    {
+        private const int height = 25;
+
+        [Resolved]
+        private OsuColour colours { get; set; }
+
+        private readonly APIKudosuHistory historyItem;
+        private readonly LinkFlowContainer linkFlowContainer;
+        private readonly DrawableDate date;
+
+        public DrawableKudosuHistoryItem(APIKudosuHistory historyItem)
+        {
+            this.historyItem = historyItem;
+
+            Height = height;
+            RelativeSizeAxes = Axes.X;
+            AddRangeInternal(new Drawable[]
+            {
+                linkFlowContainer = new LinkFlowContainer
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    AutoSizeAxes = Axes.Both,
+                },
+                date = new DrawableDate(historyItem.CreatedAt)
+                {
+                    Anchor = Anchor.CentreRight,
+                    Origin = Anchor.CentreRight,
+                }
+            });
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            date.Colour = colours.GreySeafoamLighter;
+
+            string prefix = getPrefix(historyItem);
+            var formattedSource = MessageFormatter.FormatText(getSource(historyItem));
+
+            if (!string.IsNullOrEmpty(prefix))
+            {
+                linkFlowContainer.AddText(prefix);
+                linkFlowContainer.AddText($@" {historyItem.Amount} kudosu", t =>
+                {
+                    t.Font = t.Font.With(italics: true);
+                    t.Colour = colours.Blue;
+                });
+            }
+
+            linkFlowContainer.AddLinks(formattedSource.Text + " ", formattedSource.Links);
+            linkFlowContainer.AddLink(historyItem.Post.Title, historyItem.Post.Url);
+        }
+
+        private string getSource(APIKudosuHistory historyItem)
+        {
+            string userLink() => $"[{historyItem.Giver?.Url} {historyItem.Giver?.Username}]";
+
+            switch (historyItem.Action)
+            {
+                case KudosuAction.VoteGive:
+                    return @" from obtaining votes in modding post of";
+
+                case KudosuAction.Give:
+                    return $@" from {userLink()} for a post at";
+
+                case KudosuAction.Reset:
+                    return $@"Kudosu reset by {userLink()} for the post";
```

Comment:
needto be fixed ("Kudosu" should not be in string).

Reply:
Why? Web has it like a string, for example

Reply:
kudosu is included in the other part of the string?

Reply:
![Screenshot_1](https://user-images.githubusercontent.com/22874522/64085131-728a3180-cd39-11e9-8c07-cc43f22524ba.png)
This one is an exception